### PR TITLE
Use imaplib's starttls if available

### PR DESCRIPTION
**Pull request by boehlke**
This is an approach to #217

### CODE REVIEW

File: imapclient/imapclient.py [hunk 1]
```diff
@@ -192,17 +192,20 @@ def starttls(self, ssl_context=None):
         Raises :py:exc:`AbortError` if the server does not support STARTTLS
         or an SSL connection is already established.
         """
-        if self.ssl or self._starttls_done:
-            raise self.AbortError('TLS session already established')
+        try:
+            return self._imap.starttls(ssl_context=ssl_context)
+        except AttributeError:
+            if self.ssl or self._starttls_done:
```
Comment (mjs) [inline imapclient/imapclient.py]: It's a little clearer IMO if the except clause was just a `pass` with the `if self.ssl ...` bit at the function level.

**Comment by NicolasLM** [comment]
Since we will probably have to get rid of `imaplib` at some point, I am not sure that we should rely on another of its features.

**Comment by mjs** [comment]
On reflection, I think @NicolasLM is right. We probably don't need this. Using Python's built in SSL support again has already been taken care of in a more general way. Do you agree @boehlke ?

**Comment by mjs** [comment]
Closing pending further feedback. Feel free to re-open if you think it's necessary.

**Comment by boehlke** [comment]
Hi, thanks for asking. Isn't this still an issue when using older Python versions? Please consider merging #314 ... someone might need the change. I added the change you proposed in my last commit.

**Pull request closed without merging**